src/inmanta/protocol/common.py

```diff
@@ -1158,46 +1192,212 @@ def wait(self, timeout: int = 60) -> None:
         """
         Set a callback function that is to be called when the result is ready.
         """
         self._callback = fnc
 
-    async def all(self) -> AsyncIterator[types.JsonType]:
+    def value(self) -> R:
+        """
+        Returns the value wrapped in this result, parsed as the method's return type. Only works for typed methods.
+        If paged, returns only the values for the page represented by this result. To get all results, see `all()`.
+
+        Converts return codes to http exceptions where appliccable.
+
+        :raises BaseHttpException: when return code is not 200
+        """
+        if self.code != 200:
+            exc_mapping: Mapping[int, type[exceptions.BaseHttpException]] = {
+                400: exceptions.BadRequest,
+                401: exceptions.UnauthorizedException,
+                403: exceptions.Forbidden,
+                404: exceptions.NotFound,
+                409: exceptions.Conflict,
+                500: exceptions.ServerError,
+                503: exceptions.ShutdownInProgress,
+            }
+            exception: type[exceptions.BaseHttpException] = exc_mapping.get(self.code, exceptions.ServerError)
+
+            raise (
+                exception(message=self.result.get("message", None), details=self.result.get("error_details", None))
+                if self.result is not None
+                else exception()
+            )
+
+        # typed methods always require an envelope key
+        if self.result is None or self.method_properties.envelope_key not in self.result:
+            raise exceptions.BadRequest("No data was provided in the body. Make sure to only use typed methods.")
+
+        warnings: Optional[object] = self.result.get("metadata", {}).get("warnings", None)
+        if warnings is not None:
+            if not isinstance(warnings, list):
+                raise exceptions.BadRequest("Invalid warnings metadata attached to method response.")
+            for warning in warnings:
+                if not isinstance(warning, str):
+                    raise exceptions.BadRequest("Invalid warnings metadata attached to method response.")
+                LOGGER.warning(warning)
+
+        if self.method_properties.return_type is None:
+            return None
+
+        try:
+            ta = pydantic.TypeAdapter(self.method_properties.return_type)
+        except InvalidMethodDefinition:
+            raise exceptions.BadRequest("Typed client can only be used with typed methods.")
+
+        return ta.validate_python(self.result[self.method_properties.envelope_key])
+
+    @typing.overload
+    def pageable[V: types.SimpleTypes](self: "Result[list[V]]") -> "PageableResult[V]": ...
+
+    @typing.overload
+    def pageable(self: "Result[types.SinglePageTypes]") -> typing.NoReturn: ...
+
+    @typing.overload
+    def pageable(self) -> object: ...  # catch-all for unknown parameter types and non-list sequence types
+
+    def pageable(self) -> object:
+        """
```

Comment:
I don't like the existence of this method, but out of all the many options I tried or considered, this was by far the cleanest.